exercises/grep/grep.spec.js

```diff
@@ -0,0 +1,233 @@
+        files: ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
+      })).resolves.toBe(`${resolveDataFile('paradise-lost.txt')}:4:With loss of Eden, till one greater Man`)
+    })
+
+    it('Multiple files, no matches, various flags', () => {
+      return expect(spawn_grep({
+        pattern: "Frodo",
+        flags: ["-n", "-l", "-x", "-i"],
+        files: ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
+      })).resolves.toBe('')
+    })
+
+    it('Multiple files, several matches, file flag takes precedence over line number flag', () => {
+      return expect(spawn_grep({
+        pattern: "who",
+        flags: ["-n", "-l"],
+        files: ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
+      })).resolves.toBe(`${resolveDataFile('iliad.txt')}\n${resolveDataFile('paradise-lost.txt')}`)
+    })
+
+    it('Multiple files, several matches, inverted and match entire lines flags', () => {
+      return expect(spawn_grep({
+        pattern: "Illustrious into Ades premature,",
+        flags: ["-x", "-v"],
+        files: ["iliad.txt", "midsummer-night.txt", "paradise-lost.txt"]
+      })).resolves.toBe(`${resolveDataFile('iliad.txt')}:Achilles sing, O Goddess! Peleus' son;\n${resolveDataFile('iliad.txt')}:His wrath pernicious, who ten thousand woes\n${resolveDataFile('iliad.txt')}:Caused to Achaia's host, sent many a soul\n${resolveDataFile('iliad.txt')}:And Heroes gave (so stood the will of Jove)\n${resolveDataFile('iliad.txt')}:To dogs and to all ravening fowls a prey,\n${resolveDataFile('iliad.txt')}:When fierce dispute had separated once\n${resolveDataFile('iliad.txt')}:The noble Chief Achilles from the son\n${resolveDataFile('iliad.txt')}:Of Atreus, Agamemnon, King of men.\n${resolveDataFile('midsummer-night.txt')}:I do entreat your grace to pardon me.\n${resolveDataFile('midsummer-night.txt')}:I know not by what power I am made bold,\n${resolveDataFile('midsummer-night.txt')}:Nor how it may concern my modesty,\n${resolveDataFile('midsummer-night.txt')}:In such a presence here to plead my thoughts;\n${resolveDataFile('midsummer-night.txt')}:But I beseech your grace that I may know\n${resolveDataFile('midsummer-night.txt')}:The worst that may befall me in this case,\n${resolveDataFile('midsummer-night.txt')}:If I refuse to wed Demetrius.\n${resolveDataFile('paradise-lost.txt')}:Of Mans First Disobedience, and the Fruit\n${resolveDataFile('paradise-lost.txt')}:Of that Forbidden Tree, whose mortal tast\n${resolveDataFile('paradise-lost.txt')}:Brought Death into the World, and all our woe,\n${resolveDataFile('paradise-lost.txt')}:With loss of Eden, till one greater Man\n${resolveDataFile('paradise-lost.txt')}:Restore us, and regain the blissful Seat,\n${resolveDataFile('paradise-lost.txt')}:Sing Heav'nly Muse, that on the secret top\n${resolveDataFile('paradise-lost.txt')}:Of Oreb, or of Sinai, didst inspire\n${resolveDataFile('paradise-lost.txt')}:That Shepherd, who first taught the chosen Seed`)
```

Comment:
Can you add the newlines in the string directly so it's not this long? String template literals allow for newlines!

Reply:
The problem is with the tabs, i used a function to remove the tabs. If you though about something better, let me know 👍